public/scripts/ssooverride.js

```diff
@@ -6,12 +6,19 @@
 (function() {
   var hostname = document.getElementById("ssooverride").getAttribute("data-login-hostname");
   var loginToSave = document.getElementById("ssooverride").getAttribute("data-login-to-save");
+  var oauthClientId = document.getElementById("ssooverride").getAttribute("data-oauth-clientid");
 
-  require(["jquery", "thimblePage", "url-template"], function($, editor, urlTemplate) {
-
+  require(["jquery", "thimblePage", "url-template", "uuid", "cookies"], function($, editor, urlTemplate, uuid, cookies) {
     // we chronicle login status with a "loggedin" class on the <html> tag
     var html = document.getElementsByTagName("html")[0];
 
+    function setStateCookie(state) {
+      var domain;
+
+      cookies.expire("state");
+      cookies.set("state", state);
```

Comment:
XSS is the worst threat because it means you can do anything, so I wouldn't worry about this particular problem :)